rust/lance/src/dataset/fragment.rs

```diff
@@ -944,30 +944,31 @@ impl FileFragment {
         }
     }
 
-    /// Take rows based on internal local row ids
+    /// Take rows based on internal local row offsets
     ///
-    /// If the row ids are out-of-bounds, this will return an error. But if the
-    /// row id is marked deleted, it will be ignored. Thus, the number of rows
-    /// returned may be less than the number of row ids provided.
+    /// If the row offsets are out-of-bounds, this will return an error. But if the
+    /// row offset is marked deleted, it will be ignored. Thus, the number of rows
+    /// returned may be less than the number of row offsets provided.
     ///
-    /// To recover the original row ids from the returned RecordBatch, set the
-    /// `with_row_id` parameter to true. This will add a column named `_row_id`
+    /// To recover the original row addresses from the returned RecordBatch, set the
+    /// `with_row_address` parameter to true. This will add a column named `_rowaddr`
     /// to the RecordBatch at the end.
```

Comment:
Now that we are starting to have stable IDs I wonder if the "take by offsets" method should someday go away?  I guess I can still see it being useful if the user wants to sample the dataset (they don't need to first know the row ids and can just randomly permute the offsets) but they could just grab the row ids first (saves us the work of having to do it later).

I don't actually have an opinion here, it's more an honest question to see what you think.

Reply:
I think once they are fully stable (stable after updates, not just move-stable) this seems reasonable. Though I think people like offsets too for some use-cases. From an access pattern perspective, it can make sense.